api/odl.py

```diff
@@ -1400,12 +1448,10 @@ def _get(
         patron = patron or flask.request.patron
         _db = Session.object_session(patron)
         collection = self.collection(_db)
-        shared_secret = ConfigurationSetting.for_library_and_externalintegration(
-            _db,
-            ExternalIntegration.PASSWORD,
-            patron.library,
-            collection.external_integration,
-        ).value
+        config = collection.integration_configuration
+        shared_secret = config.for_library(patron.library.id).get(
```

Comment:
While removing the getitem interface I noticed this value should never exist since the SharedODLAPI has no library settings. Is this an instance of settings inheritance?